packages/blocks/src/__internal__/utils/dom.ts

```diff
@@ -0,0 +1,304 @@
+import {
+  BLOCK_CHILDREN_CONTAINER_PADDING_LEFT,
+  BLOCK_ID_ATTR,
+} from '@blocksuite/global/config';
+import { assertExists } from '@blocksuite/global/utils';
+
+import type { Loader } from '../../components/loader.js';
+import type { BlockComponentElement, ContainerBlock } from './query.js';
+import type { Point, Rect } from './rect.js';
+
+const AFFINE_CODE = 'AFFINE-CODE';
```

Comment:
I'm thinking about merging this module with the `query.ts` in some following PRs, since their functionalities are highly overlapped, not sure how to distinguish which util should go where.

Reply:
Also, for adding a new block flavour, we have to also define it here as hard-coded constants. This is something we are trying hard to avoid.

Reply:
> I'm thinking about merging this module with the query.ts in some following PRs, since their functionalities are highly overlapped, not sure how to distinguish which util should go where.

We can merge dom.ts into query.ts.

Reply:
> Also, for adding a new block flavour, we have to also define it here as hard-coded constants. This is something we are trying hard to avoid.

No good solution has been thought of here, only a few nested types are kept for now.